src/main/java/com/google/sps/data/DataStoreManager.java

```diff
@@ -0,0 +1,242 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import java.io.IOException;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Servlet responsible for creating new tasks. **/
+public class DataStoreManager implements Datamanager {
+
+  private Datastore datastore;
+
+  public DataStoreManager (){  
+  }
+
+  /**
+    * Stores the static data of a single game in datastore as an entity
+    * @param game an Game variable representing a single instance of a game.
+  */
+  public void storeGame (Game game){
+    long timestamp = System.currentTimeMillis();
+
+    Entity gameEntity = new Entity("Game", game.getGameID());
+    gameEntity.setProperty("gameName", game.getGameName());
+    gameEntity.setProperty("gameCreator", game.getGameCreator());
+    gameEntity.setProperty("gameDescription", game.getGameDescription());
+    gameEntity.setProperty("numStages", game.getNumStages());
+    gameEntity.setProperty("stages", game.getStages());
+    gameEntity.setProperty("numTimesPlayed", game.getNumTimesPlayed());
+    gameEntity.setProperty("numTimesFinished", game.getNumTimesFinished());
+    gameEntity.setProperty("numStarVotes", game.getNumStarVotes());
+    gameEntity.setProperty("totalStars", game.getTotalStars());
+    gameEntity.setProperty("numDifficultyVotes", game.getNumDifficultyVotes());
+    gameEntity.setProperty("totalDifficulty", game.getTotalDifficulty());
+    gameEntity.setProperty("timestamp", timestamp);
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    datastore.put(gameEntity);
+  }
+  
+  /**
+    * Updates an entity of a single game data in datastore
+    * @param game an Game variable representing a single instance of a game.
+  */
+  public void updateGame(Game game){
+    long timestamp = System.currentTimeMillis();
+    Key gameEntityKey = KeyFactory.createKey("Game", game.getGameID());
+    try {
+      Entity gameEntity = datastore.get(gameEntityKey);
```

Comment:
Defining this variable inside the try block won't allow you to access it afterwards. You need to define gameEntity outside the block and set the value inside it.